File: haromszogmokas/MATLAB/real-time-gpu-NVIDIA.m
% 2D Sierpiński-háromszög GPU-val
clc;
clear;

% Háromszög csúcsai
A = [0, 0];
B = [1, 0];
C = [0.5, sqrt(3)/2];

% Iterációk száma
iterations = 10000;

% Kezdőpont véletlenszerűen
current_point = gpuArray([0.5, 0.5]);  % A pontok GPU-ra kerülnek

% Ábra beállítása
figure;
hold on;
axis equal;
xlim([-0.1, 1.1]);
ylim([-0.1, 1.1]);
title('Sierpiński-háromszög (GPU)');
xlabel('X');
ylabel('Y');

% Frissítési frekvencia
refresh_rate = 100;

% Iterációk
for i = 1:iterations
    % Véletlenszerűen kiválasztunk egy csúcsot
    random_vertex = randi(3);
    if random_vertex == 1
        target = gpuArray(A);
    elseif random_vertex == 2
        target = gpuArray(B);
    else
        target = gpuArray(C);
    end
    
    % Új pont számítása (felezőpont)
    current_point = (current_point + target) / 2;
    
    % Pont kirajzolása
    plot(current_point(1), current_point(2), 'b.', 'MarkerSize', 1);
    
    % Progresszió és ábra frissítése
    if mod(i, refresh_rate) == 0
        percent_complete = (i / iterations) * 100;
        fprintf('Progress: %.2f%%\n', percent_complete);
        drawnow;
    end
end

% Végső állapot
fprintf('Progress: 100.00%% - Done!\n');

File: haromszogmokas/Python/real-time-gpu-NVIDIA.py
import cupy as cp
import matplotlib.pyplot as plt
import random
import time

# Háromszög csúcsai
A = cp.array([0, 0])
B = cp.array([1, 0])
C = cp.array([0.5, cp.sqrt(3) / 2])

# Iterációk száma
iterations = 10000

# Kezdőpont véletlenszerűen
current_point = cp.array([0.5, 0.5])

# Ábra beállítása
plt.figure()
plt.axis('equal')
plt.xlim(-0.1, 1.1)
plt.ylim(-0.1, 1.1)
plt.title('Sierpiński-háromszög (GPU)')
plt.xlabel('X')
plt.ylabel('Y')

# Frissítési frekvencia (hány pont kirajzolása után frissítjük a progressziót)
refresh_rate = 100

# Iterációk
start_time = time.time()
for i in range(1, iterations + 1):
    # Véletlenszerűen kiválasztunk egy csúcsot
    random_vertex = random.choice([A, B, C])
    
    # Új pont számítása (felezőpont)
    current_point = (current_point + random_vertex) / 2
    
    # Pont kirajzolása
    plt.plot(current_point[0], current_point[1], 'b.', markersize=1)
    
    # Progresszió és ábra frissítése
    if i % refresh_rate == 0:
        percent_complete = (i / iterations) * 100
        elapsed_time = time.time() - start_time
        print(f"Progress: {percent_complete:.2f}% - Elapsed Time: {elapsed_time:.2f} seconds")
        plt.pause(0.001)  # Valós idejű frissítés

# Végső állapot
print("Progress: 100.00% - Done!")
plt.show()
